File: daily_problems/2025/03/0325/personal_submission/cf488b_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/convolution"
// #include "atcoder/dsu"
// #include "atcoder/fenwicktree"
// #include "atcoder/lazysegtree"
// #include "atcoder/math"
// #include "atcoder/maxflow"
// #include "atcoder/mincostflow"
// #include "atcoder/modint"
// #include "atcoder/scc"
// #include "atcoder/segtree"
// #include "atcoder/string"
// #include "atcoder/twosat"
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;
using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divided, I conquered!

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<int> nums(n);
    for (auto &x: nums) cin >> x;

    sort(nums.begin(), nums.end());

    if (n == 0) cout << "YES\n1\n1\n3\n3";
    else if (n == 1) {
        int x = nums[0];
        cout << "YES\n" << x << '\n' << 3 * x << '\n' << 3 * x;
    }
    else if (n == 2) {
        int x = nums[0], y = nums[1];
        if (x * 3 >= y)
            cout << "YES\n" << 4 * x - y << '\n' << 3 * x;
        else cout << "NO";
    }
    else if (n == 3) {
        int x = nums[0], y = nums[1], z = nums[2];
        if (z == 3 * x) cout << "YES\n" << x + z - y;
        else if (3 * x >= z && y + z == 4 * x) cout << "YES\n" << 3 * x;
        else if (z % 3 == 0 && x + y == z / 3 * 4) cout << "YES\n" << z / 3;
        else cout << "NO";
    }
    else
        cout << (nums[0] * 3 == nums[3] && nums[1] + nums[2] == nums[0] + nums[3] ? "YES" : "NO");

    return 0;
}

File: daily_problems/2025/03/0325/personal_submission/cf488b_yawn_sean.py
# Submission link: https://codeforces.com/contest/488/submission/312260566
def main():
    n = II()
    nums = [II() for _ in range(n)]
    nums.sort()

    if n == 0:
        print('YES')
        print(1, 1, 3, 3, sep='\n')
    elif n == 1:
        x = nums[0]
        print('YES')
        print(x, 3 * x, 3 * x, sep='\n')
    elif n == 2:
        x = nums[0]
        y = nums[1]
        if x * 3 >= y:
            print('YES')
            print(4 * x - y, 3 * x, sep='\n')
        else:
            print('NO')
    elif n == 3:
        x = nums[0]
        y = nums[1]
        z = nums[2]
        
        if z == 3 * x:
            print('YES')
            print(x + z - y)
        elif 3 * x >= z and y + z == 4 * x:
            print('YES')
            print(3 * x)
        elif z % 3 == 0 and x + y == z // 3 * 4:
            print('YES')
            print(z // 3)
        else:
            print('NO')
    else:
        if nums[0] * 3 == nums[3] and nums[1] + nums[2] == nums[0] + nums[3]:
            print('YES')
        else:
            print('NO')